src/bin/alert_worker.rs

```diff
@@ -13,23 +22,15 @@ async fn main() {
 
     tracing::subscriber::set_global_default(subscriber).expect("setting default subscriber failed");
 
-    let args: Vec<String> = env::args().collect();
-    // user can pass the path to a config file, but it is optional.
-    // if not provided, we use the default config.default.yaml
-    // the user also needs to pass the name of the alert stream to process
-    // stream name comes first, optional config file comes second
-    if args.len() < 2 {
-        error!("Usage: alert_worker <stream_name> <config_file>, where config_file is optional");
-        return;
-    }
+    let args = Cli::parse();
 
     let interrupt_flag = Arc::new(Mutex::new(false));
     worker_util::sig_int_handler(Arc::clone(&interrupt_flag)).await;
 
-    let stream_name = &args[1];
+    let stream_name = &args.stream.unwrap();
```

Comment:
maybe we should read those before creating that interrupt flag? That way if the values are incorrect we can catch that earlier

Reply:
I moved these up.